app/api/tax.py

```diff
@@ -43,9 +65,38 @@ class Meta:
                          type_='event')
 
 
-class TaxList(ResourceList):
+class TaxListPost(ResourceList):
     def before_post(self, args, kwargs, data):
         require_relationship(['event'], data)
+        if not has_access('is_coorganizer', event_id=data['event']):
+            raise ForbiddenException({'source': ''}, 'Co-organizer access is required.')
+
+    def before_get(self, args, kwargs):
+        if has_access('is_admin'):
+            self.schema = TaxSchema
+        else:
+            self.schema = TaxSchemaPublic
+
+    schema = TaxSchema
```

Comment:
But there is one issue. The current_identitiy will be null until jwt_required is called. So to make it work I think you have to provide `@jwt_required ` at the top of before_post method.